src/test/ui/enum-discriminant/niche-prefer-zero.rs

```diff
@@ -0,0 +1,13 @@
+// Check that niche selection prefers zero.
+// See https://github.com/rust-lang/rust/pull/87794
+// run-pass
+pub enum Size {
```

Comment:
Will probably need to `repr(u8)` here. If my memory serves me right there are some architectures that have a minimum "default" enum size which is not u8.

Reply:
*Mind blown*
Makes sense though